x-pack/plugins/apm/server/lib/services/get_services/get_services_items_stats.ts

```diff
@@ -246,3 +300,38 @@ export const getEnvironments = async ({
     environments: bucket.environments.buckets.map((env) => env.key as string),
   }));
 };
+
+export const getHealthStatuses = async (
+  { setup }: AggregationParams,
+  mlAnomaliesEnvironment?: string
+) => {
+  if (!setup.ml) {
+    return [];
+  }
+
+  const jobIds = await getMLJobIds(
+    setup.ml.anomalyDetectors,
+    mlAnomaliesEnvironment
+  );
+  if (!jobIds.length) {
+    return [];
+  }
+
+  const anomalies = await getServiceAnomalies({
+    setup,
+    environment: mlAnomaliesEnvironment,
+  });
+
+  return Object.keys(anomalies.serviceAnomalies).reduce<
+    Array<{ serviceName: string; severity?: Severity }>
+  >((prev, serviceName) => {
+    const stats = anomalies.serviceAnomalies[serviceName];
+
+    const severity = getSeverity(stats.anomalyScore);
+
+    return prev.concat({
+      serviceName,
+      severity,
+    });
+  }, []);
```

Comment:
I think I commented on something similar recently: can you rewrite with `[].map`? 

To explain why I think this is useful: `map/filter` should always be preferred over `reduce`. With `map`, without reading your implementation, I know that an array with the same length will be returned. With `filter` I know that the items will be identical (although the length might change).
`reduce` comes with barely any guarantees, since mostly anything can be returned. This forces the reader to spent more time understanding the implementation because it's not possible to assume much.

Reply:
> map/filter should always be preferred over reduce

Not sure I agree with this. Maybe when transforming and array to another array.

> reduce comes with barely any guarantees, since mostly anything can be returned

What does that mean? In this case I know an `Array<{ serviceName: string; severity?: Severity }>` is being returned.

Reply:
> Not sure I agree with this. Maybe when transforming and array to another array.

But that's exactly it. When transforming an array to another array and keeping the same length, that's what map is for.

Reply:
Fwiw, this is an oversight on my part. It should have been a .map. I think I just default to reduce out of habit when using Object.keys().